### PR TITLE
Keep a list of MAC filters installed per VF, and re-use them instead

### PR DESCRIPTION
**Pull request by stephen-hurd**
of constantly allocating more.

Also, re-enable rte_pmd_bnxt_set_all_queues_drop_en() and
rte_pmd_bnxt_set_vf_vlan_anti_spoof().

### CODE REVIEW

File: drivers/net/bnxt/rte_pmd_bnxt.c [hunk 1]
```diff
@@ -497,13 +493,29 @@ int rte_pmd_bnxt_mac_addr_add(uint8_t port, struct ether_addr *addr,
...
 
-	filter.fw_l2_filter_id = UINT64_MAX;
-	filter.flags = HWRM_CFA_L2_FILTER_ALLOC_INPUT_FLAGS_PATH_RX;
-	filter.enables = HWRM_CFA_L2_FILTER_ALLOC_INPUT_ENABLES_L2_ADDR |
+	STAILQ_FOREACH(filter, &bp->pf.vf_info[vf_id].filter, next) {
+		if (filter->flags == HWRM_CFA_L2_FILTER_ALLOC_INPUT_FLAGS_PATH_RX
+		    && filter->enables == (HWRM_CFA_L2_FILTER_ALLOC_INPUT_ENABLES_L2_ADDR
+		        | HWRM_CFA_L2_FILTER_ALLOC_INPUT_ENABLES_L2_ADDR_MASK)
+		    && memcmp(addr, filter->l2_addr, ETHER_ADDR_LEN) == 0) {
+RTE_LOG(ERR, PMD, "Re-using filter...\n");
```
Comment (ajitkhaparde) [inline drivers/net/bnxt/rte_pmd_bnxt.c]: Did you intend to delete these? The Tab-space makes me feel that. :-)

Comment (stephen-hurd) [reply drivers/net/bnxt/rte_pmd_bnxt.c]: Whoops, yeah.

File: drivers/net/bnxt/rte_pmd_bnxt.c [hunk 2]
```diff
@@ -482,7 +478,7 @@ int rte_pmd_bnxt_mac_addr_add(uint8_t port, struct ether_addr *addr,
 	/* If the VF currently uses a random MAC, update default to this one */
 	if (bp->pf.vf_info[vf_id].random_mac) {
 		if (rte_pmd_bnxt_get_vf_rx_status(port, vf_id) <= 0) {
-			bnxt_hwrm_func_vf_mac(bp, vf_id, (uint8_t *)addr);
+			rc = bnxt_hwrm_func_vf_mac(bp, vf_id, (uint8_t *)addr);
```
Comment (ajitkhaparde) [inline drivers/net/bnxt/rte_pmd_bnxt.c:481]: If this fails, shouldn't we bailout?
Also curious.. do we need to set the filter for the MAC which is also being set as default_MAC ?

Comment (stephen-hurd) [reply drivers/net/bnxt/rte_pmd_bnxt.c:481]: If we bail out, the default MAC won't be delivered to the VF at all.  By not bailing out, we ensure that even if we are unable to set the default MAC, that MAC will be delivered to the function as requested.

As for setting a filter for the default MAC, it seems that someone has to install one.  Usually the VF driver will do so, and it's not needed, but that's not guaranteed.  At some point, we may be able to save a filter by not installing one for the default MAC, but the correct choice isn't clear yet, so the conservative choice was made.

**Review by ajitkhaparde — APPROVED** [review]

**Comment by ajitkhaparde** [comment]
looks good.

**Pull request merged**